channel_type.go

```diff
@@ -1,30 +1,63 @@
 package stream_chat
 
-import "net/http"
+import (
+	"net/http"
+	"time"
+)
 
 type ChannelType struct {
+	Name             string
+	TypingEvents     bool
+	ReadEvents       bool
+	ConnectEvents    bool
+	Search           bool
+	Reactions        bool
+	Replies          bool
+	Mutes            bool
+	MessageRetention string
+	MaxMessageLength int
+	Automod          string
+	AutomodBehaviour string
+	Commands         []interface{}
+	Permissions      []interface{}
+	CreatedAt        time.Time
+	UpdatedAt        time.Time
+}
+
+// TODO:
+func (c ChannelType) toMap() map[string]interface{} {
+	return nil
 }
 
 // CreateChannelType adds new channel type
-func (c *Client) CreateChannelType(data map[string]interface{}) (err error) {
-	if data["commands"] == "" {
-		data["commands"] = "all"
+func (c *Client) CreateChannelType(chType ChannelType) (err error) {
+	if len(chType.Commands) == 0 {
+		chType.Commands = []interface{}{"all"}
 	}
-	err = c.makeRequest(http.MethodPost, "channeltypes", nil, data, nil)
 
-	return
+	err = c.makeRequest(http.MethodPost, "channeltypes", nil, chType.toMap(), nil)
+
+	return err
 }
 
 // GetChannelType returns information about channel type
-func (c *Client) GetChannelType(chanType string) (resp map[string]interface{}, err error) {
-	err = c.makeRequest(http.MethodGet, "channeltypes/"+chanType, nil, nil, &resp)
+func (c *Client) GetChannelType(chanType string) (ChannelType, error) {
+	var resp struct {
+		ChannelTypes map[string]ChannelType `json:"channel_types"`
+	}
 
-	return
+	err := c.makeRequest(http.MethodGet, "channeltypes/"+chanType, nil, nil, &resp)
+
+	return resp.ChannelTypes[chanType], err
```

Comment:
Check presence with error.

Reply:
API will return 404 it it's not exists. But let's do not trust API :)